numpy/lib/tests/test_utils.py

```diff
@@ -56,11 +56,24 @@ def test_safe_eval_nameconstant():
     utils.safe_eval('None')
 
 
-def test_byte_bounds():
-    a = arange(12).reshape(3, 4)
-    low, high = utils.byte_bounds(a)
-    assert_equal(high - low, a.size * a.itemsize)
-
+class TestByteBounds(object):
+
+    def test_byte_bounds(self):
+        a = arange(12).reshape(3, 4)
+        low, high = utils.byte_bounds(a)
+        assert_equal(high - low, a.size * a.itemsize)
+
+    def test_non_contiguous_positive_stride(self):
```

Comment:
I'd argue that this array, and the one below, is contiguous, just in an unusual order. You should add a test for strided arrays with padding, perhaps using `[::2]`